src/models/state/mempool.rs

```diff
@@ -152,6 +162,7 @@ impl Mempool {
     pub fn new(
         max_total_size: ByteSize,
         max_num_transactions: Option<usize>,
+        max_num_inputs_per_tx: Option<usize>,
```

Comment:
This interface is starting to get unwieldy. How about a builder instead? If you don't want to write one yourself, take a look at [bon](https://crates.io/crates/bon).